packages/expo-build-properties/src/ios.ts

```diff
@@ -0,0 +1,66 @@
+import { IOSConfig, ConfigPlugin, withXcodeProject, XcodeProject } from '@expo/config-plugins';
+
+import type { PluginConfigType } from './pluginConfig';
+
+const { createBuildPodfilePropsConfigPlugin } = IOSConfig.BuildProperties;
+
+export const withIosBuildProperties = createBuildPodfilePropsConfigPlugin<PluginConfigType>(
+  [
+    {
+      propName: 'ios.useFrameworks',
+      propValueGetter: (config) => config.ios?.useFrameworks,
+    },
+  ],
+  'withIosBuildProperties'
+);
+
+export const withIosDeploymentTarget: ConfigPlugin<PluginConfigType> = (config, props) => {
+  const deploymentTarget = props.ios?.deploymentTarget;
+  if (!deploymentTarget) {
+    return config;
+  }
+
+  // Updates deployment target in app xcodeproj
+  config = withIosDeploymentTargetXcodeProject(config, { deploymentTarget });
+
+  // Updates deployement target in Podfile (Pods project)
+  config = withIosDeploymentTargetPodfile(config, props);
+
+  return config;
+};
+
+const withIosDeploymentTargetXcodeProject: ConfigPlugin<{ deploymentTarget: string }> = (
+  config,
+  props
+) => {
+  return withXcodeProject(config, (config) => {
+    config.modResults = updateDeploymentTargetXcodeProject(
+      config.modResults,
+      props.deploymentTarget
+    );
+    return config;
+  });
+};
+
+function updateDeploymentTargetXcodeProject(
+  project: XcodeProject,
+  deploymentTarget: string
+): XcodeProject {
+  const configurations: Record<string, any> = project.pbxXCBuildConfigurationSection();
```

Comment:
This may have issues at scale when a project supports something like App Clips where the minimum deployment target must can only go as low as `15.4`. Maybe add a note about providing some future property to apply only to `PBXNativeTarget`, `productType`s that match `com.apple.product-type.application`. The complexity seems out of scope for this PR, but it seems worth noting.

Reply:
added in https://github.com/expo/expo/pull/17106/commits/8cd3f67dfba0f84b8def0a20e573e24eef01a620